x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/RealmSettings.java

```diff
@@ -29,7 +36,24 @@
  */
 public class RealmSettings {
 
-    public static final String RESERVED_REALM_NAME_PREFIX = "_";
+    public static final Setting.AffixSetting<List<String>> DOMAIN_TO_REALM_ASSOC_SETTING = Setting.affixKeySetting(
+        "xpack.security.authc.domains.",
+        "realms", // TODO prevent domains with_ (and check that the synthetic realms use this, __attach, account tokens)
+        key -> Setting.stringListSetting(key, new Setting.Validator<>() {
+            @Override
+            public void validate(List<String> realms) {
+                for (String realm : realms) {
+                    if (realm.startsWith(RESERVED_REALM_AND_DOMAIN_NAME_PREFIX)) {
+                        throw new IllegalArgumentException(
+                            "Realm names must not start with \"" + RESERVED_REALM_AND_DOMAIN_NAME_PREFIX + "\""
```

Comment:
I think we can drop this validation entirely because:
* Realm names and deprecation are done elsewhere
* Domain and realm cross validation is also done elsewhere
* Realm name with leading underscore is not fully blocked so it is still a valid, though no-ideal, use case.

Reply:
Okay, removed. I would think that because this is a new feature we could limit it to non-deprecated realms only, as a way to nudge realm renaming.